### PR TITLE
Update flow-bin

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: core/remote/cozy.js [hunk 1]
```diff
@@ -219,7 +219,7 @@ class RemoteCozy {
   }
 
   // Retrieve the path of a remote file doc
-  async _setPath (doc /*: any */) /*: Promise<void> */ {
+  async _setPath (doc /*: * */) /*: Promise<*> */ {
```
Comment (taratatach) [inline core/remote/cozy.js]: Can't we use `RemoteDoc` to type `doc`?

File: core/remote/document.js [hunk 2]
```diff
@@ -91,7 +91,7 @@ export type JsonApiDoc = {
 }
 */
 
-function jsonApiToRemoteDoc (json/*: JsonApiDoc */) /*: RemoteDoc */ {
+function jsonApiToRemoteDoc (json/*: JsonApiDoc */) /*: * */ {
```
Comment (taratatach) [inline core/remote/document.js]: What are we missing to type the return value?

File: core/remote/cozy.js (hunk 1 above)
Comment (nono) [reply core/remote/cozy.js]: Nope, it is called with things that looks like `RemoteDoc` but are not: they don't have a `path`, that is mandatory for `RemoteDoc`

File: core/app.js [hunk 3]
```diff
@@ -140,12 +140,12 @@ class App {
...
-      if (err.code === 'ENOTFOUND') {
-        log.warn(`The DNS resolution for ${parsed.hostname} failed.`)
-        log.warn('Are you sure the domain is OK?')
-      } else if (err === 'Bad credentials') {
+      if (err === 'Bad credentials') {
         log.warn(err)
         log.warn('Are you sure there are no typo on the passphrase?')
+      } else if (err.code === 'ENOTFOUND') {
+        log.warn(`The DNS resolution for ${parsed.hostname} failed.`)
+        log.warn('Are you sure the domain is OK?')
```
Comment (sebn) [inline core/app.js]: Out of curiosity, why did the order have to change?

File: core/local/chokidar_event.js [hunk 4]
```diff
@@ -2,12 +2,13 @@
...
 import type fs from 'fs'
+import type {Metadata} from '../metadata'
 
 export type ChokidarAdd = {type: 'add', path: string, stats: fs.Stats}
 export type ChokidarAddDir = {type: 'addDir', path: string, stats: fs.Stats}
 export type ChokidarChange = {type: 'change', path: string, stats: fs.Stats}
-export type ChokidarUnlink = {type: 'unlink', path: string}
-export type ChokidarUnlinkDir = {type: 'unlinkDir', path: string}
+export type ChokidarUnlink = {type: 'unlink', path: string, old?: Metadata}
+export type ChokidarUnlinkDir = {type: 'unlinkDir', path: string, old?: Metadata}
```
Comment (sebn) [inline core/local/chokidar_event.js]: It's weird that you had to do the [exact same change](https://github.com/cozy-labs/cozy-desktop/pull/1275#discussion-diff-233448601L9) as @aenario in #1275

Comment (sebn) [reply core/local/chokidar_event.js]: Ok, got it.

Could you add a FIXME explaining that the offline unlink events detection is abusing the ChokidarUnlink* event types which actually never have an old property?

File: core/app.js (hunk 3 above)
Comment (nono) [reply core/app.js]: Flow complained that we can try to access `err.code` when `err` can be the string `'Bad credentials'`

File: core/local/chokidar_event.js (hunk 4 above)
Comment (sebn) [reply core/local/chokidar_event.js]: @nono Forget about it, I'll fix it in a separate PR.

Comment (nono) [reply core/local/chokidar_event.js]: What is really weird is that flow didn't tell us about that before

**Comment by sebn** [comment]
:+1:

File: core/remote/document.js (hunk 2 above)
Comment (taratatach) [reply core/remote/document.js]: Arf OK. I suppose this is the same issue as above.

**Pull request merged**